frappe/__init__.py

```diff
@@ -1106,14 +1090,7 @@ def _respond(doc, from_redis=False):
 
 
 def _set_document_in_cache(key: str, doc: "Document") -> None:
-	local.document_cache[key] = doc
-
-	# Avoid setting in local.cache since we're already using local.document_cache above
-	# Try pickling the doc object as-is first, else fallback to doc.as_dict()
-	try:
-		cache().hset("document_cache", key, doc, cache_locally=False)
-	except Exception:
-		cache().hset("document_cache", key, doc.as_dict(), cache_locally=False)
```

Comment:
> any reason why we're not going to have the as_dict fallback?

This is something Ankush & I had discussed as would-do in next major release. Mainly because **Simple is better than Complex**. And that dict caches aren't that performant.

Almost all doctypes are pickle-able (except Quickbooks Migrator - can be fixed, is never supposed to be cached anyway). That a Document should be pickle-able when loaded is a very reasonable assumption IMO.

Reply:
the changes, though look good to me, i don't have the context of the discussion. but it does make sense. i'll merge this.

(again apologies for the stupid assumption :P - just wrote what i thought first off the top of my head)

<details>
<summary>also, we should probably look into transaction handling for `get_cached_doc` ..the cache updation (clearing happens post save) only happens when get_doc is called</summary>
![Screenshot from 2023-01-25 21-49-45](https://user-images.githubusercontent.com/32034600/214620417-2278350c-7768-448a-9fe8-d8e0515c0889.png)
</details>